server/src/main/java/org/server/SupServer.java

```diff
@@ -112,4 +131,78 @@ public void tellSomeone(String message, String fromname, String toname) {
             } catch (Exception ex) { ex.printStackTrace(); }
         }
     }
-}
\ No newline at end of file
+    
+    /**
+     * return the status to the specific one
+	 *
+	 * @param 
+	 * 	status - status info
+	 *  toname - the specific one to transfer status
+     * */
+    public void tellSomeoneStatus(String status, String toname)
+    {
+    	System.out.println("Return the \"" + status +"\" to " + toname);
+    	if (Contacts.getInstance().hasContact(toname)) {
+            try {
+                PrintWriter writer = Contacts.getInstance().getContact(toname);
+                writer.println(status);
+                writer.flush();
+            } catch (Exception ex) { ex.printStackTrace(); }
+        }
+    }
+    
+    /**
+     * return the status to everyone
+	 *
+	 * @param 
+	 * 	status - status info
+     * */
+	public void tellEveryoneStatus(String status) {
+		System.out.println("Return the \"" + status + "\" to all");
+		try {
+			
+			List<PrintWriter> list = Contacts.getInstance().getWriterList();
+
+			for (PrintWriter writer : list) {
+				writer.println(status);
+				writer.flush();
+			}
+		} catch (Exception ex) {
+			ex.printStackTrace();
+		}
+	}
+	
+	/**
+     * remove contact from active list and send the online list to everyone
+	 *
+	 * @param 
+	 * 	name - the contact removed from the list
+     * */
+	public void removeContact(String name)
+	{
+		Contacts.getInstance().removeContact(name);
+		System.out.println(name + "has been removed from the contact");
+	}
+	
+	/**
+     * send user list after remove someone
+	 *
+     * */
+	public void sendList()
```

Comment:
this is good, we might have to disable it until @danielalvarez0 implements the contact list functionality though, our client won't know what to do with it yet!